context.go

```diff
@@ -44,20 +44,15 @@ func (c Context) Array(key string, arr LogArrayMarshaler) Context {
 		c.l.context = arr.write(c.l.context)
 		return c
 	}
-	var a *Array
-	if aa, ok := arr.(*Array); ok {
```

Comment:
This code is unreachable, and also did nothing. The test is the same as the one [line 43](https://github.com/rs/zerolog/pull/737/changes#diff-552f47512a00afe5fc6850cc9ddc830a6daeca162750e50aab4ed549685e0253R43) so it would never be true. That means we're actually always going to execute `a = Arr()` and fall down to the write.